quickwit/quickwit-indexing/src/source/ingest/mod.rs

```diff
@@ -135,6 +137,7 @@ pub struct IngestSource {
     fetch_stream: MultiFetchStream,
     publish_lock: PublishLock,
     publish_token: PublishToken,
+    shard_positions: FnvHashMap<ShardId, Position>,
```

Comment:
Why not reuse the `assigned_shards: HashMap<ShardId, AssignedShard>` to track the publish position?

Reply:
ah yes that's perfect! I am glad I had forgotten to put that PR in draft  :)